packages/features/ee/workflows/components/WorkflowStepContainer.tsx

```diff
@@ -919,9 +918,31 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
                       )}
                   </div>
                 )}
-
-                {step.action !== WorkflowActions.SMS_ATTENDEE &&
-                step.action !== WorkflowActions.SMS_NUMBER ? (
+                {!isEmailSubjectNeeded && !isSenderIsNeeded && (
+                  <>
+                    <div className="mb-2 flex items-center pb-[1.5px]">
+                      <Label className="mb-0 flex-none ">
+                        {isEmailSubjectNeeded ? t("email_body") : t("text_message")}
+                      </Label>
+                    </div>
+                    <Editor
+                      getText={() => {
+                        return props.form.getValues(`steps.${step.stepNumber - 1}.reminderBody`) || "";
+                      }}
+                      setText={(text: string) => {
+                        props.form.setValue(`steps.${step.stepNumber - 1}.reminderBody`, text);
+                        props.form.clearErrors();
+                      }}
+                      variables={DYNAMIC_TEXT_VARIABLES}
+                      height="200px"
+                      updateTemplate={updateTemplate}
+                      firstRender={firstRender}
+                      setFirstRender={setFirstRender}
+                      editable={false}
+                    />
+                  </>
+                )}
+                {isEmailSubjectNeeded && (
```

Comment:
This is not optimal to use a similar editor and render it conditionally. Instead, make it non-editable when certain conditions are met

Reply:
The issue with the code is `!props.readOnly && !isWhatsappAction(step.action)` value is `false` still I am able to edit the text box. Hence I separated it out and set it to false.

I can create a new reusable component called EditorComponent and take isEditable as param which will be set true for email case and false for Whatsapp

Reply:
Something of this sort 

```

const editorComponent = ({ isEditable }) => {
      return (
        <>
          <div className="mb-2 flex items-center pb-[1.5px]">
            <Label className="mb-0 flex-none ">
              {isEmailSubjectNeeded ? t("email_body") : t("text_message")}
            </Label>
          </div>
          <Editor
            getText={() => {
              return props.form.getValues(`steps.${step.stepNumber - 1}.reminderBody`) || "";
            }}
            setText={(text: string) => {
              props.form.setValue(`steps.${step.stepNumber - 1}.reminderBody`, text);
              props.form.clearErrors();
            }}
            variables={DYNAMIC_TEXT_VARIABLES}
            height="200px"
            updateTemplate={updateTemplate}
            firstRender={firstRender}
            setFirstRender={setFirstRender}
            editable={isEditable}
          />
        </>
      );
    };







               
          
      {!isEmailSubjectNeeded && !isSenderIsNeeded && <editorComponent isEditable={false} />}
      {isEmailSubjectNeeded && <editorComponent isEditable={true} />}

Reply:
Hey @Flash-Shivam, I've tested it locally, and currently, there is very weird behavior. If you hardcode isEditable to false, it’s not editable, but when it's set using a variable or certain conditions, it’s still editable. Let me investigate why this is happening, and then I will suggest a fix.

Reply:
I feel if we can't resolve the issue of why it is editable when set false using a variable or certain conditions, we should for time being use my approach of creating a editor component and use the conditions outside it and pass the isEditable value as true or false instead of conditions.